include/nbl/builtin/hlsl/cpp_compat/impl/intrinsics_impl.hlsl

```diff
@@ -6,425 +6,762 @@
-		return spirv::findILsb(val);
-#else
-		return glm::findLSB(val);
-#endif
+		using UnsignedInteger = typename hlsl::unsigned_integer_of_size_t<sizeof(Integer)>;
+		return std::popcount(static_cast<UnsignedInteger>(val));
 	}
 };
 
-template<>
-struct find_lsb_helper<uint32_t>
+template<typename SquareMatrix>
+requires concepts::Matrix<SquareMatrix> && matrix_traits<SquareMatrix>::Square
+struct inverse_helper<SquareMatrix>
 {
-	static int32_t findLSB(NBL_CONST_REF_ARG(uint32_t) val)
+	static SquareMatrix __call(NBL_CONST_REF_ARG(SquareMatrix) mat)
 	{
-#ifdef __HLSL_VERSION
-		return spirv::findILsb(val);
-#else
-		return glm::findLSB(val);
-#endif
+		return reinterpret_cast<SquareMatrix&>(glm::inverse(reinterpret_cast<typename SquareMatrix::Base const&>(mat)));
 	}
 };
 
-#define DEFINE_FIND_LSB_COMMON_SPECIALIZATION(INPUT_INTEGER_TYPE, INTEGER_TYPE)\
-template<>\
-struct find_lsb_helper<INPUT_INTEGER_TYPE>\
-{\
-	static int32_t findLSB(NBL_CONST_REF_ARG(INPUT_INTEGER_TYPE) val)\
-	{\
-		return find_lsb_helper<INTEGER_TYPE>::findLSB(val);\
-	}\
-};\
+template<typename EnumT>
+requires std::is_enum_v<EnumT>
+struct bitCount_helper<EnumT>
+{
+	using return_t = int32_t;
+	using underlying_t = std::underlying_type_t<EnumT>;
+	static return_t __call(NBL_CONST_REF_ARG(EnumT) val)
+	{
+		return bitCount_helper<const underlying_t>::__call(reinterpret_cast<const underlying_t&>(val));
+	}
+};
 
-DEFINE_FIND_LSB_COMMON_SPECIALIZATION(int16_t, int32_t)
-DEFINE_FIND_LSB_COMMON_SPECIALIZATION(uint16_t, uint32_t)
-#ifndef __HLSL_VERSION
-DEFINE_FIND_LSB_COMMON_SPECIALIZATION(int8_t, int32_t)
-DEFINE_FIND_LSB_COMMON_SPECIALIZATION(uint8_t, uint32_t)
-#endif
+template<typename T, typename U>
+requires concepts::FloatingPoint<T> && (concepts::FloatingPoint<T> || concepts::Boolean<T>)
+struct mix_helper<T, U>
+{
+	using return_t = T;
+	static inline return_t __call(const T x, const T y, const U a)
+	{
+		return glm::mix(x, y, a);
+	}
+};
 
-template<>
-struct find_lsb_helper<uint64_t>
+template<typename T>
+requires concepts::FloatingPointScalar<T> || concepts::IntegralScalar<T>
+struct sign_helper<T>
 {
-	static int32_t findLSB(NBL_CONST_REF_ARG(uint64_t) val)
+	using return_t = T;
+	static inline return_t __call(const T val)
 	{
-#ifdef __HLSL_VERSION
-		const uint32_t lowBits = uint32_t(val);
-		const int32_t lowLsb = find_lsb_helper<uint32_t>::findLSB(lowBits);
+		if (val < 0)
+			return -1;
+		if (val > 0)
+			return 1;
 
-		if (lowLsb == -1)
-		{
-			const uint32_t highBits = uint32_t(val >> 32);
-			const int32_t highLsb = find_lsb_helper<uint32_t>::findLSB(highBits);
-			if (highLsb == -1)
-				return -1;
-			else
-				return 32 + highLsb;
-		}
+		return 0;
+	}
+};
 
-		return lowLsb;
-#else
-		return glm::findLSB(val);
-#endif
+template<typename T>
+requires concepts::FloatingPointScalar<T>
+struct radians_helper<T>
+{
+	using return_t = T;
+	static inline return_t __call(const T degrees)
+	{
+		return degrees * (numbers::pi<T> / static_cast<T>(180.0));
```

Comment:
I think `pi` needs to be `bit_cast<T>` before usage because its an integer bitpattern ( blame no static inline float in DXC / C++ until C++20)